impl/src/main/java/org/ehcache/impl/internal/store/heap/OnHeapStore.java

```diff
@@ -1233,30 +1234,35 @@ public V apply(final K k, final V oldValue) {
     return storeEventDispatcher;
   }
 
-  private OnHeapValueHolder<V> setAccessTimeAndExpiryThenReturnMapping(K key, OnHeapValueHolder<V> valueHolder, long now) {
+  private OnHeapValueHolder<V> setAccessTimeAndExpiryThenReturnMappingOutsideLock(K key, OnHeapValueHolder<V> valueHolder, long now) {
     Duration duration;
     try {
       duration = expiry.getExpiryForAccess(key, valueHolder.value());
     } catch (RuntimeException re) {
       LOG.error("Expiry computation caused an exception - Expiry duration will be 0 ", re);
+      duration = Duration.ZERO;
+    }
+    valueHolder.accessed(now, duration);
+    if (Duration.ZERO.equals(duration)) {
       expireMapping(key, valueHolder);
       return null;
     }
-    valueHolder.accessed(now, duration);
     return valueHolder;
   }
 
-  private OnHeapValueHolder<V> setAccessTimeAndExpiryThenReturnMapping(K key, OnHeapValueHolder<V> valueHolder, long now,
+  private OnHeapValueHolder<V> setAccessTimeAndExpiryThenReturnMappingUnderLock(K key, OnHeapValueHolder<V> valueHolder, long now,
```

Comment:
The UnderLock and OutsideLock variants of this method are identical (practically so). Why having two variants then?